workflow/react-project/src/components/Header/index.jsx

```diff
@@ -1,18 +1,25 @@
 import { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { faMartiniGlassCitrus } from '@fortawesome/free-solid-svg-icons';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import { faStar } from '@fortawesome/free-regular-svg-icons';
+import { faHouseUser } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { changeModalKey } from '@store';
 
 import Button from '@common/Button';
-import Modal from '@common/Modal';
+import SignUp from '../Modals/Authentication/SignUp';
+import SignIn from '../Modals/Authentication/SignIn';
 
+import { changeModalKey } from '@store/appConfigurations/action';
+import { selectAuthenticationToken } from '@store/authentication/selectors';
 import './style.less';
 
 const Header = () => {
   const dispatch = useDispatch();
+  const authenticationToken = useSelector(selectAuthenticationToken);
```

Comment:
you don't need authenticationToken here in header
for displaying data you need to know whether user is logged in or not.
Create selector `isUserLoggedInSelector` and use variable `isUserLoggedIn`